lib/swaps/SwapClientManager.ts

```diff
@@ -330,25 +330,38 @@ class SwapClientManager extends EventEmitter {
    * @param currency a currency that should be linked with a swap client.
    * @returns Nothing upon success, throws otherwise.
    */
-  public add = (currency: Currency): void => {
-    if (currency.swapClient === SwapClientType.Raiden && currency.tokenAddress && this.raidenClient) {
-      this.swapClients.set(currency.id, this.raidenClient);
-      this.raidenClient.tokenAddresses.set(currency.id, currency.tokenAddress);
-      this.emit('raidenUpdate', this.raidenClient.tokenAddresses, this.raidenClient.address);
+  public add = async (currency: Currency) => {
+    if (currency.tokenAddress) {
+      if (currency.swapClient === SwapClientType.Connext) {
+        if (!this.connextClient) {
+          throw errors.SWAP_CLIENT_NOT_CONFIGURED(currency.id);
+        }
+        this.swapClients.set(currency.id, this.connextClient);
+        this.connextClient.tokenAddresses.set(currency.id, currency.tokenAddress);
+        this.emit('connextUpdate', this.connextClient.tokenAddresses, this.connextClient.address);
+      } else if (currency.swapClient === SwapClientType.Raiden) {
```

Comment:
Maybe we should remove the raiden part since it's never going to be used and it would make the function more readable?

Reply:
Fixed in https://github.com/ExchangeUnion/xud/pull/1746/commits/4b113412c8afcb24df56ef6bec955acc4a33d357#diff-8aac1fd5a8463c73f6951a8eb02f5f15.